### ISSUE TITLE
BrainScript output NAN value

### ISSUE THREAD

**Issue by cwbforever** [Microsoft/CNTK#2689]
I always obtain NAN value when testing the model, trained by brainscript. The errs is also weird. I trained the model many times with different parameters but the errs value is always the same (85.154%).

Final Results: Minibatch[1-2]: errs = 85.154% * 7140; ce = **nan** * 7140; perplexity = **nan**

vocabSize = 18935 ; numHosts = 100
TrainTagger = {
    action = "train"
    BrainScriptNetworkBuilder = {
        inputDim  = $vocabSize$
        intentDim = $numHosts$
        embDim = 500
        hiddenDim = 300

        BiRecurrentLSTMLayer {outDim} = {
            F = RecurrentLSTMLayer {outDim, goBackwards=false}
            G = RecurrentLSTMLayer {outDim, goBackwards=true}
            apply (x) = Splice (BS.Sequences.Last(F(x)):BS.Sequences.First(G(x)))
        }.apply

        model = Sequential (
            EmbeddingLayer {embDim} :
            BatchNormalizationLayer {} :
            BiRecurrentLSTMLayer {hiddenDim} :
            BatchNormalizationLayer {} :
            DenseLayer {intentDim}                      ##### different dimension
        )

        # features
        n = DynamicAxis()
        document     = Input {inputDim, dynamicAxis=n}
        hostLabels   = Input {intentDim}

        # model application
        z = model (document)

        # loss and metric
        ce   = CrossEntropyWithSoftmax (hostLabels, z)
        errs = ClassificationError     (hostLabels, z)
        p = Softmax (z)

        featureNodes    = (document)
        labelNodes      = (hostLabels)
        criterionNodes  = (ce)
        evaluationNodes = (errs)
        outputNodes     = (p)
    }

    SGD = {
        maxEpochs = 60 ; epochSize = 60000

        minibatchSize = 70

        learningRatesPerSample = 0.005*5:0.003*10:0.002
        gradUpdateType = "fsAdaGrad"
        gradientClippingWithTruncation = true ; clippingThresholdPerSample = 15.0

        firstMBsToShowResult = 10 ; numMBsToShowResult = 100
        ParallelTrain = {
            parallelizationMethod = "DataParallelSGD"
            parallelizationStartEpoch = 1
            distributedMBReading = true
            dataParallelSGD = { gradientBits = 2 }
        }
    }

    reader = {
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/train.ctf"
        randomize = true
        input = {
            document   = { alias = "S0" ; dim = $vocabSize$ ;  format = "sparse" }
            hostLabels = { alias = "S1" ; dim = $numHosts$ ; format = "sparse" }
        }
    }
}

**Comment by ebarsoumMS** [comment]
If you simplify the network will you still get a nan? What is your dataset?

**Comment by cwbforever** [comment]
Yes, I use a simplified network by removing a BatchNormalizationLayer and then the results look normal. What is the root cause? 

model = Sequential (
        EmbeddingLayer {embDim} :
        BatchNormalizationLayer {} :
        BiRecurrentLSTMLayer {hiddenDim} :
        **#** BatchNormalizationLayer {} :
        DenseLayer {intentDim}                      ##### different dimension
    )

Here is the data used to train the model (very similar to ATIS).
0	|S0 1967:1 |# BOS	|S1 86:1 |# dailystar.co.uk
0	|S0 17061:1 |# tiemoue
0	|S0 1230:1 |# bakayoko
0	|S0 13263:1 |# pulls
0	|S0 314:1 |# agony
0	|S0 2920:1 |# chelsea
0	|S0 17304:1 |# training
0	|S0 184:1 |# adds
0	|S0 8404:1 |# injury
0	|S0 18650:1 |# woe
0	|S0 5621:1 |# EOS
2	|S0 1967:1 |# BOS	|S1 56:1 |# telegraph.co.uk
2	|S0 11426:1 |# nigerian
2	|S0 12738:1 |# police
2	|S0 15215:1 |# shoot
2	|S0 6794:1 |# gang
2	|S0 9159:1 |# kidnapped
2	|S0 2178:1 |# britons
2	|S0 5621:1 |# EOS
4	|S0 1967:1 |# BOS	|S1 1:1 |# foxnews.com
4	|S0 18934:1 |# bigfoot[UNKNOWN]
4	|S0 13948:1 |# reportedly
4	|S0 15322:1 |# sighted
4	|S0 11510:1 |# northern
4	|S0 2452:1 |# california
4	|S0 12517:1 |# pictures
4	|S0 18125:1 |# viral
4	|S0 5621:1 |# EOS
